web/src/ShareSnapshotModal.tsx

```diff
@@ -0,0 +1,59 @@
+import React from "react"
+import { Snapshot } from "./types"
+import "./ShareSnapshotModal.scss"
+
+type props = {
+  handleSendSnapshot: (snapshot: Snapshot) => void
+  handleClose: () => void
+  show: boolean
+  state: Snapshot
+  snapshotUrl: string
+  registerTokenUrl: string
+}
+export default function ShareSnapshotModal(props: props) {
+  const showHideClassName = props.show
+    ? "modal display-block"
+    : "modal display-none"
+  const hasLink = props.snapshotUrl !== ""
+  if (!hasLink) {
+    return (
+      <div className={showHideClassName}>
+        <section className="modal-main">
+          <section className="modal-snapshotUrlWrap">
+            <button onClick={props.handleClose}>close</button>
+            <button onClick={() => props.handleSendSnapshot(props.state)}>
```

Comment:
it seems weird to pass modal the entire state. usually you would bind the snapshot to the function, so that this was just invoked as props.sendSnapshot()